File: trunk/java/UMLGraph/src/org/daisy/urakawa/media/asset/PlainTextMediaAsset.java
package org.daisy.urakawa.media.asset;

import org.daisy.urakawa.xuk.XmlDataReader;
import org.daisy.urakawa.xuk.XmlDataWriter;

import java.nio.charset.Charset;

/**
 *
 */
public class PlainTextMediaAsset implements MediaAsset {
    public Charset getEncoding() {return null;}

    public void setEncoding(Charset cs) {}

    /**
     * @hidden
     */
    public String getUid() {
        return null;
    }

    /**
     * @hidden
     */
    public String getName() {
        return null;
    }

    /**
     * @hidden
     */
    public void setName(String name) {
    }

    /**
     * @hidden
     */
    public void setAssetManager(MediaAssetManager man) {
    }

    /**
     * @hidden
     */
    public MediaAssetManager getAssetManager() {
        return null;
    }

    /**
     * @hidden
     */
    public DataProvider getDataProvider() {
        return null;
    }

    /**
     * @hidden
     */
    public boolean XukIn(XmlDataReader source) {
        return false;
    }

    /**
     * @hidden
     */
    public boolean XukOut(XmlDataWriter destination) {
        return false;
    }

    /**
     * @hidden
     */
    public String getXukLocalName() {
        return null;
    }

    /**
     * @hidden
     */
    public String getXukNamespaceURI() {
        return null;
    }
}

File: branches/convenience-views/implementation/UrakawaToolkit/urakawa/media/asset/PlainTextMediaAsset.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace urakawa.media.asset
{
	public class PlainTextMediaAsset : IMediaAsset
	{
		public Encoding getEncoding();
		public void setEncoding(Encoding enc);

		#region IMediaAsset Members

		public IMediaAssetManager getAssetManager()
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public string getUid()
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public string getName()
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public void setName(string newName)
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public IDataProvider getDataProvider()
		{
			throw new Exception("The method or operation is not implemented.");
		}

		#endregion

		#region IXukAble Members

		public bool XukIn(System.Xml.XmlReader source)
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public bool XukOut(System.Xml.XmlWriter destination)
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public string getXukLocalName()
		{
			throw new Exception("The method or operation is not implemented.");
		}

		public string getXukNamespaceUri()
		{
			throw new Exception("The method or operation is not implemented.");
		}

		#endregion
	}
}
